### PR TITLE
feat: liquidity provisions view

### LINKED ISSUE #491: Liquidity providers view
design from console
Image
- [ ] check if all columns are needed
- [ ] check for other feedback from team/users
- [ ] find a place to put it in Console V2
- [ ] Liquidity data that applies to the whole market (not specific provisions)
- [ ] Consider including the market.account.liquidity balances which is the sum of fees to be distributed in the next liquidity epoch 

See https://github.com/vegaprotocol/frontend-monorepo/issues/625 as there is some liquidity info that was left out of the market info panel. e..g liquidity monitoring target stake etc.
Screenshot 2022-06-23 at 10 25 15
if these don't go into this ticket, please raise another.

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #491 & #626

# Description ℹ️

Liquidity provisions view launched from market info. 

# Demo 📺

If user is LP: 
Screenshot 2022-09-01 at 23 16 36

If user is not LP:
Screenshot 2022-09-01 at 23 18 12

If wallet not connected:
image

Open LP view 1:
Screenshot 2022-09-01 at 16 48 14

Open LP view 2:
Screenshot 2022-09-01 at 16 48 27

# Technical 👨‍🔧

- [x] view full oracle details
- [x] view liquidity provisions view

### CODE REVIEW

File: libs/ui-toolkit/src/components/dialog/dialog.tsx [hunk 1]
```diff
@@ -37,8 +37,9 @@ export function Dialog({
     {
       'lg:w-[620px] w-full': size === 'small',
       'w-full w-full md:w-[720px] lg:w-[940px]': size === 'medium',
+      'w-full w-full md:w-[80vw] lg:w-[80vw]': size === 'large',
       'left-[0px] top-[99px] h-[calc(100%-99px)] border-0 translate-x-[0] translate-y-[0] border-none overflow-y-auto w-full':
-        size === 'large',
+        size === 'extra-large',
```
Comment (MadalinaRaicu) [inline libs/ui-toolkit/src/components/dialog/dialog.tsx:42]: I needed something in between `medium` and `large`. I am super open to suggestions.

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 2]
```diff
@@ -131,28 +139,31 @@ export const Info = ({ market }: InfoProps) => {
...
         title: t(`Insurance pool`),
         content: (
           <MarketInfoTable
             data={{
-              balance: `${a.balance}
-           ${market.tradableInstrument.instrument.product?.settlementAsset.symbol}`,
+              balance: a.balance,
             }}
+            assetSymbol={assetSymbol}
+            decimalPlaces={market.positionDecimalPlaces} // TODO decimal places needed here?
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:150]:
```suggestion
            decimalPlaces={market.positionDecimalPlaces}
```

File: libs/orders/src/lib/components/order-feedback/order-feedback.tsx [hunk 3]
```diff
@@ -52,12 +52,11 @@ export const OrderFeedback = ({ transaction, order }: OrderFeedbackProps) => {
                 : 'text-vega-red-dark dark:text-vega-red'
             }
           >
-            {`${
-              order.side === Side.SIDE_BUY ? '+' : '-'
-            } ${addDecimalsFormatNumber(
-              order.size,
-              order.market?.positionDecimalPlaces ?? 0
-            )}
+            {`${order.side === Side.SIDE_BUY ? '+' : '-'
```
Comment (dexturr) [inline libs/orders/src/lib/components/order-feedback/order-feedback.tsx:55]: nitpick (non-blocking): seeing this pop up in a few places over the last little while. May be worth a util.

File: libs/deal-ticket/src/components/market-info/tooltip-mapping.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { t } from '@vegaprotocol/react-helpers';
+import { Link } from '@vegaprotocol/ui-toolkit';
+import type { ReactNode } from 'react';
+
+export const tooltipMapping: Record<string, ReactNode> = {
+  makerFee: t(
+    'Maker portion of the fee is transferred to the non-aggressive, or passive party in the trade (the maker, as opposed to the taker).'
+  ),
+  liquidityFee: t(
+    'Liquidity portion of the fee is paid to market makers for providing liquidity, and is transferred to the market-maker fee pool for the market.'
```
Comment (candida-d) [inline libs/deal-ticket/src/components/market-info/tooltip-mapping.tsx:10]:
```suggestion
    'Liquidity portion of the fee is paid to market makers for providing liquidity, and is transferred to the liquidity fee pool for the market.'
```

File: libs/liquidity/src/lib/__generated__/MarketLiquidity.ts [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  /**
+   * The name of the asset (string)
+   */
+  settlementAsset: MarketLiquidity_market_tradableInstrument_instrument_product_settlementAsset;
+}
+
+export interface MarketLiquidity_market_tradableInstrument_instrument {
+  __typename: "Instrument";
+  /**
+   * A short non necessarily unique code used to easily describe the instrument (e.g: FX:BTCUSD/DEC18) (string)
```
Comment (candida-d) [inline libs/liquidity/src/lib/__generated__/MarketLiquidity.ts:121]:
```suggestion
   * A short not necessarily unique code used to easily describe the instrument (e.g: FX:BTCUSD/DEC18) (string)
```

File: libs/deal-ticket/src/components/market-info/tooltip-mapping.tsx (hunk 4 above)
Comment (candida-d) [reply libs/deal-ticket/src/components/market-info/tooltip-mapping.tsx:10]: @MadalinaRaicu This is not coming from generated docs, correct?

Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/tooltip-mapping.tsx:10]: correct. it has been manually added in another ticket.

File: libs/deal-ticket/src/components/market-info/info-market-query.ts [hunk 6]
```diff
@@ -8,13 +8,6 @@ export const MARKET_INFO_QUERY = gql`
       decimalPlaces
       positionDecimalPlaces
       state
-      accounts {
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/market-info/info-market-query.ts:11]: removing these duplicates - might have been added during a merge conflict

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 7]
```diff
@@ -285,172 +336,3 @@ export const Info = ({ market }: InfoProps) => {
     </div>
   );
 };
-
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:288]: this file was getting too big. moved the tooltips somewhere else.

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        inline={isPrimitive}
+        noBorder={true}
+        dtClassName={className}
+        ddClassName={className}
+      >
+        <Tooltip description={tooltipMapping[field]} align="start">
+          <div tabIndex={-1}>{startCase(t(field))}</div>
+        </Tooltip>
+        <span style={{ wordBreak: 'break-word' }}>
+          {isNumber && !unformatted
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:52]: chore (blocking): Can we tidy up this huge condition. Preferably set the cell content in the body of the function? This is hard to read
```
let content = null
if (x) {
  content = ...
} else {
  content =
}
return <div>{content}</div>
```

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+} from '@vegaprotocol/ui-toolkit';
+import BigNumber from 'bignumber.js';
+import omit from 'lodash/omit';
+import startCase from 'lodash/startCase';
+import type { ReactNode } from 'react';
+import { tooltipMapping } from './tooltip-mapping';
+
+interface RowProps {
+  field: string;
+  value: any;
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:21]: chore (blocking): can we type this properly.

It looks like we aren't using generating types for the MARKET_INFO_QUERY. Can we do so please. Then all the types should flow nicely through this part of the app.

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+            : value}
+        </span>
+      </KeyValueTableRow>
+    );
+  }
+  return null;
+};
+
+export interface MarketInfoTableProps {
+  data: any;
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:70]: chore (blocking): any

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+  value,
+  decimalPlaces,
+  asPercentage,
+  unformatted,
+  assetSymbol = '',
+}: RowProps) => {
+  const isNumber = typeof value === 'number' || !isNaN(Number(value));
+  const isPrimitive = typeof value === 'string' || isNumber;
+  const className = 'text-black dark:text-white text-ui !px-0 !font-normal';
+  if (isPrimitive) {
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:39]: question (non blocking): Not sure why we need this check. I think it stems from some any types defined in this file.

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 12]
```diff
@@ -51,7 +49,10 @@ export const MarketInfoContainer = ({ marketId }: MarketInfoContainerProps) => {
   }, []);
 
   const { data, loading, error } = useQuery(MARKET_INFO_QUERY, {
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:51]: chore (blocking): I realize this was already this way.. but can we add generic type args here please

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 13]
```diff
@@ -51,7 +49,10 @@ export const MarketInfoContainer = ({ marketId }: MarketInfoContainerProps) => {
...
-    variables: { marketId, interval: Interval.INTERVAL_I1H, since: yTimestamp },
+    variables: useMemo(
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:52]: chore (suggestion): can we keep hooks used naturally in the body of the function for readability

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 14]
```diff
@@ -128,28 +131,34 @@ export const Info = ({ market }: InfoProps) => {
...
-  );
+  const { VEGA_EXPLORER_URL } = useEnvironment();
+  const keyDetails = {
+    ...pick(
+      market,
+      'name',
+      'decimalPlaces',
+      'positionDecimalPlaces',
+      'tradingMode',
+      'id' as 'marketId'
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:158]: question (blocking): Why the typecast, doesnt look like it does anything?

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 15]
```diff
@@ -254,6 +270,32 @@ export const Info = ({ market }: InfoProps) => {
...
+          }}
+          decimalPlaces={
+            market.tradableInstrument.instrument.product.settlementAsset
+              .decimals
+          }
+          assetSymbol={assetSymbol}
+          link={
+            <Link
+              target="_blank"
+              className="text-ui dark:text-white text-black underline hover:underline hover:text-black-60 dark:hover:text-white-80"
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:290]: chore (blocking): None of these classes are required, and some are from the old config. Lets just use the raw Link component styles which should handle underline styles

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 16]
```diff
@@ -267,6 +309,15 @@ export const Info = ({ market }: InfoProps) => {
               market.tradableInstrument.instrument.product
                 .oracleSpecForTradingTermination.id,
           }}
+          link={
+            <Link
+              target="_blank"
+              href={`${VEGA_EXPLORER_URL}/oracles#${market.tradableInstrument.instrument.product.oracleSpecForTradingTermination.id}`}
+              className="text-ui dark:text-white text-black underline hover:underline hover:text-black-60 dark:hover:text-white-80"
```
Comment (mattrussell36) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:316]: chore (blocking): Remove classnames please. As an aside. We should have to set text color everywhere on every element. Those styles should be inherited

File: libs/liquidity/src/lib/liquidity-data-provider.ts [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+
+export interface LiquidityProvision {
+  party: string;
+  commitmentAmount: string | undefined;
+  fee: string | undefined;
+  equityLikeShare: string;
+  averageEntryValuation: string;
+  obligation: string | null;
+  supplied: string | null;
+  status: LiquidityProvisionStatus | string | undefined;
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-data-provider.ts:88]: question (blocking): Why type this as also string | undefined? The api indicates that this can only every be `LiquidityProvisionStatus`. Some thing is wrong in the types elsewhere if you've had to do this. If its because node can be undefined then you should filter those nodes out first.

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+  partyId,
+  marketId,
+}: LiquidityManagerProps) => {
+  const gridRef = useRef<AgGridReact | null>(null);
+  const {
+    data: marketLiquidityData,
+    loading,
+    error,
+  } = useQuery<MarketLiquidity>(MARKET_LIQUIDITY_QUERY, {
+    variables: useMemo(() => ({ marketId }), [marketId]),
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:39]: chore (blocking): No need for useMemo here

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+  });
+
+  const { liquidityProviders, suppliedStake, targetStake, code, symbol } =
+    useLiquidityProvision({ data: marketLiquidityData });
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:43]: suggestion (non blocking): IMO I'd keep the query and the data manipulation all in one hook. So components that need to use the data can consume it more simply

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+    'h-full max-h-full',
+    'flex flex-col',
+    'text-ui-small',
+    'gap-8'
+  );
+  const titleClasses =
+    'sm:text-lg md:text-xl lg:text-2xl flex items-center gap-4 whitespace-nowrap';
+
+  const myLpEdges = useMemo(
+    () => (liquidityProviders || []).filter((e) => e.party === partyId),
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:55]: suggestion (non blocking): The result of the useLiquidityProvision hook should always be an array so that these falsy checks can be omitted.

File: libs/liquidity/src/lib/liquidity-data-provider.ts [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+export const formatLiquidityData = ({
+  suppliedStake,
+  assetDecimalPlaces,
+  targetStake,
+  code,
+  symbol,
+  liquidityProviders,
+}: LiquidityData) => {
+  return {
+    suppliedStake: addDecimalsFormatNumber(suppliedStake, assetDecimalPlaces),
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-data-provider.ts:185]: suggestion (non blocking): Its better to return the raw numbers/bignumbers rather than format them here. What if another component wants to format them in a different way?

Utilize AgGrid valueFormatters instead.

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+    variables: useMemo(() => ({ marketId }), [marketId]),
+  });
+
+  const { liquidityProviders, suppliedStake, targetStake, code, symbol } =
+    useLiquidityProvision({ data: marketLiquidityData });
+
+  const wrapperClasses = classNames(
+    'h-full max-h-full',
+    'flex flex-col',
+    'text-ui-small',
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:48]: chore (blocking): Remove this has no effect. use text-xs/sm/md etc

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+    if (activeEdges?.length) return 'active';
+    else if (inactiveEdges?.length > 0) return 'inactive';
+    return 'active';
+  };
+  const itemClass =
+    'min-w-min w-[120px] whitespace-nowrap pb-3 px-4 border-l border-neutral-300 dark:border-neutral-700';
+  const itemHeading = 'text-neutral-400';
+  return (
+    <AsyncRenderer loading={loading} error={error} data={liquidityProviders}>
+      <header className="w-screen xl:px-4 pt-4 border-b border-neutral-300 dark:border-neutral-700">
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:83]: chore (blocking): This UI should be the concern of the app and not the lib as it deals with page layout. Ideally the lib should only export things that are app agnostic. In this case only the table component and the data provider/hook should be exported from the lib. Things like market id and party id should be passed in as props from the app.

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+              <Tabs activeDefaultId="myLP">
+                <Tab id="myLP" name={t('My liquidity provision')}>
+                  <LiquidityTable ref={gridRef} data={myLpEdges} />
+                </Tab>
+                <Tab children={undefined} id={''} name={''} hidden={true} />
+              </Tabs>
+            </ResizableGridPanel>
+          )}
+          <ResizableGridPanel preferredSize={300} minSize={50}>
+            <Tabs activeDefaultId={getActiveDefaultId()}>
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:119]: chore (blocking): can we rename this prop to just `active`

File: libs/react-helpers/src/lib/format/number.ts [hunk 25]
```diff
@@ -19,11 +19,11 @@ export function toBigNum(
 
 export function addDecimal(
   value: string | number | EthersBigNumber,
-  decimals: number,
+  decimals?: number,
```
Comment (mattrussell36) [inline libs/react-helpers/src/lib/format/number.ts:22]: question (blocking): I'm uncomfortable with this change. This function is used a lot to add and remove decimals specifically for price and sizes. Its important that decimals is ALWAYS provided so that this can be done accurately. No someone could easily omit this which could result in transactions being sent with incorrect values.

File: libs/react-helpers/src/lib/format/number.ts [hunk 26]
```diff
@@ -55,19 +55,25 @@ export const formatNumber = (
 };
 
 export const addDecimalsFormatNumber = (
-  rawValue: string | number,
-  decimalPlaces: number,
+  rawValue?: string | number | null,
```
Comment (mattrussell36) [inline libs/react-helpers/src/lib/format/number.ts:58]: question (blocking): I'm also uncomfortable with these changes. These are simple formatting functions. If you need to pass undefined here something is probably wrong with your types and/or code elsewhere

File: libs/liquidity/src/lib/liquidity-manager.tsx [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+              <div style={{ wordBreak: 'break-word' }}>{marketId}</div>
+            </div>
+          </div>
+        </div>
+      </header>
+      <div className={wrapperClasses}>
+        <ResizableGrid vertical={true}>
+          {partyId && (
+            <ResizableGridPanel preferredSize={300} minSize={50}>
+              <Tabs activeDefaultId="myLP">
```
Comment (mattrussell36) [inline libs/liquidity/src/lib/liquidity-manager.tsx:110]: chore (blocking): Why use Tabs component if there is only one tab? I feel like it might be because you want the tab title UI above the table. 

Lets keep it more simple and just render a My LP tab alongside active and inactive

File: libs/react-helpers/src/lib/format/date.ts [hunk 28]
```diff
@@ -31,6 +31,13 @@ export const getDateTimeFormat = once(
     })
 );
 
+export const dateValueFormatter = ({ value }: { value?: string | null }) => {
```
Comment (mattrussell36) [inline libs/react-helpers/src/lib/format/date.ts:34]: suggestion (non blocking):  I'm not sure we really need extra formatter. Just use getDateTimeFormat().format()?

File: libs/web3/src/lib/use-ethereum-config.ts [hunk 29]
```diff
@@ -32,6 +32,19 @@ export const NETWORK_PARAMS_QUERY = gql`
   }
 `;
 
+export const useNetworkParameter = (params: string[]) => {
```
Comment (mattrussell36) [inline libs/web3/src/lib/use-ethereum-config.ts:35]: chore (blocking): API now allows for getting a single param so lets refactor and use that.

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx (hunk 8 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-key-value-table.tsx]: sure, updated

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 15 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-market.tsx]: yeah, happy to remove them now. updated

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 14 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-market.tsx:158]: it's not a typecast, it changes the key. Instead of displaying 'ID' in this section, it will display 'Market ID'. 
I followed this approach because I was trying to not miss any field from the market info that the API was giving me, maybe omit some controlled things like `__typename`, but also add new ones. this is why I was checking primitives and formatting keys. 
it makes it **super** easy to update the market info section. 
literally, one line of code is needed to control the fields. 
this seemed necessary when I was working on the market info section and I was going back and forth to update things. the list was getting super large, hundreds of lines of code, formatting seemed to be spread around, and it was repetitive. 
I am happy to discuss how to improve this component. this is why we were not following MarketInfoQuery types fully, .

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 16 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-market.tsx]: yeah, happy to remove them now. updated

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 18 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: updated.

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 20 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: updated.

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 22 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: yep, now these can be removed. needed them for the previous design.

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 24 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: updated

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 27 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: yeah, I am happy to do that. that's what I was proposing yesterday. check this out: 
Screenshot 2022-09-02 at 10 33 49
Screenshot 2022-09-01 at 23 46 54
updated.

File: libs/react-helpers/src/lib/format/date.ts (hunk 28 above)
Comment (MadalinaRaicu) [reply libs/react-helpers/src/lib/format/date.ts]: updated.

File: libs/liquidity/src/lib/liquidity-data-provider.ts (hunk 21 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-data-provider.ts]: updated. before using only ag-grid for this, I had to change the formatting in two places. now I don;t need to do this anymore. moved the value formatters back.

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 19 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: nice! updated.

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 12 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-market.tsx:51]: updated.

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 13 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-market.tsx:52]: updated.

File: libs/web3/src/lib/use-ethereum-config.ts (hunk 29 above)
Comment (MadalinaRaicu) [reply libs/web3/src/lib/use-ethereum-config.ts]: updated. updated also in token app to use web3 lib. the methods were duplicated. hope that is fine.

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx (hunk 11 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:51]: https://github.com/vegaprotocol/frontend-monorepo/pull/1133/files#r961402382

File: libs/react-helpers/src/lib/format/number.ts (hunk 25 above)
Comment (MadalinaRaicu) [reply libs/react-helpers/src/lib/format/number.ts]: updated.

File: libs/react-helpers/src/lib/format/number.ts (hunk 26 above)
Comment (MadalinaRaicu) [reply libs/react-helpers/src/lib/format/number.ts]: updated.

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx (hunk 9 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:20]: https://github.com/vegaprotocol/frontend-monorepo/pull/1133/files#r961402382

File: libs/deal-ticket/src/components/market-info/info-key-value-table.tsx (hunk 10 above)
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/info-key-value-table.tsx:72]: https://github.com/vegaprotocol/frontend-monorepo/pull/1133/files#r961402382

File: libs/liquidity/src/lib/liquidity-manager.tsx (hunk 23 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-manager.tsx]: updated

File: libs/liquidity/src/lib/liquidity-data-provider.ts (hunk 17 above)
Comment (MadalinaRaicu) [reply libs/liquidity/src/lib/liquidity-data-provider.ts]: for every data.liquidityProviderFeeShare we are trying to find the matching liquidity provision connection node. the API does not match it for us. I tried to filter non-empty connection nodes, however typescript being typescript still complains if I leave the connection as not being possibly null/undefined.

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 16 above)
Comment (mattrussell36) [reply libs/deal-ticket/src/components/market-info/info-market.tsx]: Sorry this should have read. "We should _not_ have to set text color everywhere..."

File: libs/deal-ticket/src/components/market-info/info-market.tsx (hunk 14 above)
Comment (mattrussell36) [reply libs/deal-ticket/src/components/market-info/info-market.tsx:161]: It is a typecast. All its doing here is typing the last argument you pass to `pick` as the literal string 'marketId' as opposed to it just being inferred as `string`. Its safe to remove here because the label text actually comes from the `marketID` key passed into the MarketInfoTable.

Comment (mattrussell36) [reply libs/deal-ticket/src/components/market-info/info-market.tsx:161]: Overall I think that rather than trying to ensure all the properties on the object are displayed we should more explicitly declare what fields we are showing. This should improve the type safety and simplify things quite a lot as its quite hard to follow all the data and additional props for formatting.

Comment (mattrussell36) [reply libs/deal-ticket/src/components/market-info/info-market.tsx]: That said, I think lets merge this and revisit it

**Review by mattrussell36 — APPROVED** [review]

File: apps/trading/components/trading-mode-tooltip/trading-mode-tooltip.tsx [hunk 30]
```diff
@@ -65,8 +63,7 @@ const compileGridData = (market: Market_market) => {
       label: isLiquidityMonitoringAuction
         ? t('Est auction end')
         : t('Auction end'),
-      value: getDateTimeFormat().format(new Date(market.data.auctionEnd)),
-      isEstimate: isLiquidityMonitoringAuction ? true : false,
```
Comment (notbot00) [inline apps/trading/components/trading-mode-tooltip/trading-mode-tooltip.tsx:69]: question (blocking?): not sure if this was a change in requirements here? are we now displaying all market types prefixed with a `~`?

**Review by notbot00 — APPROVED** [review]

File: libs/web3/src/lib/use-ethereum-config.ts [hunk 31]
```diff
@@ -32,6 +32,32 @@ export const NETWORK_PARAMS_QUERY = gql`
   }
 `;
 
+export const useNetworkParam = (param: string) => {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/web3/src/lib/use-ethereum-config.ts]: updated.

File: libs/deal-ticket/src/components/market-info/index.ts [hunk 32]
```diff
@@ -0,0 +1,3 @@
+export * from './__generated__';
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/deal-ticket/src/components/market-info/index.ts:1]: happy to do this as a follow up.

File: apps/trading/pages/liquidity/[marketId].page.tsx [hunk 33]
```diff
@@ -0,0 +1,93 @@
...
+  const inactiveEdges = useMemo(
+    () =>
+      liquidityProviders.filter(
+        (e) => e.status !== LiquidityProvisionStatus.STATUS_ACTIVE
+      ),
+    [liquidityProviders]
+  );
+
+  const getActiveDefaultId = () => {
+    if (myLpEdges?.length > 0) return 'myLP';
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/trading/pages/liquidity/[marketId].page.tsx]: updated.

File: apps/trading/components/trading-mode-tooltip/trading-mode-tooltip.tsx (hunk 30 above)
Comment (MadalinaRaicu) [reply apps/trading/components/trading-mode-tooltip/trading-mode-tooltip.tsx:69]: @mattrussell36 removed it here https://github.com/vegaprotocol/frontend-monorepo/pull/1133/commits/57bcec59d8e02c6b71ca3ad6bde3a0369113b525  was this in the design? perhaps we should check with @campbellssource

Comment (mattrussell36) [reply apps/trading/components/trading-mode-tooltip/trading-mode-tooltip.tsx:69]: Good catch. I've amended it to include the tilde when in its a liq monitoring auction

File: libs/deal-ticket/src/components/market-info/info-market.tsx [hunk 34]
```diff
@@ -137,39 +137,45 @@ export const Info = ({ market }: InfoProps) => {
...
+    ...pick(
+      market,
+      'name',
+      'decimalPlaces',
+      'positionDecimalPlaces',
+      'tradingMode'
+    ),
+    state: MarketStateMapping[market.state],
+  };
+  console.log(keyDetails);
```
Comment (MadalinaRaicu) [inline libs/deal-ticket/src/components/market-info/info-market.tsx:167]:
```suggestion
```
